### PR TITLE
Rename BusNotifications to Notifications

### LINKED ISSUE #3425: Rename BusNotifications
With the decision to remove "bus" from our APIs, we need to rename `BusNotifications` to something more appropriate.

Maybe we can already align the naming to the [proposed notifications vision](https://github.com/Particular/PlatformDevelopment/issues/605).

### PR DESCRIPTION
**Pull request by timbussmann**
fix #3425

Rename `BusNotifications` to `Notifications`

I moved over the obsolete property to `Notifications` to provide more help to migrating users. Should we do that or keep it on the obsolete class?

WIP doc PR: https://github.com/Particular/docs.particular.net/pull/1342

### CODE REVIEW

**Comment by andreasohlund** [comment]
>I moved over the obsolete property to Notifications to provide more help to migrating users. Should we do that or keep it on the obsolete class?

Seems reasonable, when going with the event aggregator we'll obsolete the entire `Notification` class anyway so keeping a few obsoletes there seems like a small price to pay

File: src/NServiceBus.AcceptanceTests/Recoverability/Retries/When_performing_slr_and_counting.cs [hunk 1]
```diff
@@ -50,16 +50,14 @@ public RetryEndpoint()
             class ErrorNotificationSpy : IWantToRunWhenBusStartsAndStops
             {
                 Context testContext;
-                BusNotifications notifications;
+                Notifications notifications;
```
Comment (andreasohlund) [inline src/NServiceBus.AcceptanceTests/Recoverability/Retries/When_performing_slr_and_counting.cs]: Should we reformat this file to put the variables at the bottom or is that out of scope?

**Comment by andreasohlund** [comment]
Looks good, @bording want to give it a once over and merge?

File: src/NServiceBus.AcceptanceTests/Recoverability/Retries/When_performing_slr_and_counting.cs (hunk 1 above)
Comment (timbussmann) [inline src/NServiceBus.AcceptanceTests/Recoverability/Retries/When_performing_slr_and_counting.cs]: I made a note about cleaning up the tests since there are also many which need namespace cleanup. Will make a dedicated PR.

**Comment by bording** [comment]
:+1:
